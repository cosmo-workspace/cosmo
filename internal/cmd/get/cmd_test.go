package get

import (
	"bytes"
	"testing"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/spf13/cobra"
)

func TestCommandGet(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl get suite")
}

var _ = Describe("help", func() {
	It("should match snapshot", func() {
		cmd := &cobra.Command{}
		out := bytes.Buffer{}
		cmd.SetOut(&out)
		AddCommand(cmd, cli.NewRootOptions())
		cmd.SetArgs([]string{"get", "--help"})
		err := cmd.Execute()
		Expect(err).ShouldNot(HaveOccurred())
		Expect(out.String()).To(MatchSnapShot())
	})
})
