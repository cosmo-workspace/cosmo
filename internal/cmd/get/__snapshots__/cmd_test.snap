['help should match snapshot 1']
SnapShot = """
Get cosmo resources

Usage:
   get [command]

Available Commands:
  events      Get events for user
  network     Get workspace networks
  template    Get workspace templates
  user        Get users. Alias of 'cosmoctl user get'
  useraddon   Get user addons. Alias of 'cosmoctl user get-addons'
  workspace   Get workspaces. Alias of 'cosmoctl workspace get'

Flags:
  -h, --help   help for get

Use \" get [command] --help\" for more information about a command.
"""
